File: Core-backend-layer/Programming-fundamentals/Python/DesignPatterns/src/memento/reservation_memento.py
# Memento class to hold the reservation state
class ReservationMemento:
    def __init__(self, customer_name, party_size, date_time):
        self.customer_name = customer_name
        self.party_size = party_size
        self.date_time = date_time

    def get_customer_name(self):
        return self.customer_name

    def get_party_size(self):
        return self.party_size

    def get_date_time(self):
        return self.date_time

File: Core-backend-layer/Programming-fundamentals/Java/DesignPatterns/src/memento/ReservationMemento.java
package memento;

// Memento class to hold the reservation state
public class ReservationMemento {
    private final String customerName;
    private final int partySize;
    private final String dateTime;

    public ReservationMemento(String customerName, int partySize, String dateTime) {
        this.customerName = customerName;
        this.partySize = partySize;
        this.dateTime = dateTime;
    }

    public String getCustomerName() {
        return customerName;
    }

    public int getPartySize() {
        return partySize;
    }

    public String getDateTime() {
        return dateTime;
    }
}